### PR TITLE
Fixes for julia 0.7, drop 0.5 support

### PR DESCRIPTION
**Pull request by carlobaldassi**
Tests pass without deprecations on my machine.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,15 +3,15 @@ os:
...
   - nightly
 notifications:
   email: false
 # uncomment the following lines to override the default test script
-# script:
-#  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-#  - julia --check-bounds=yes -e 'Pkg.clone(pwd()); Pkg.build("AxisAlgorithms"); Pkg.test("AxisAlgorithms"; coverage=true)'
+script:
+ - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
+ - julia --check-bounds=yes --inline=no -e 'Pkg.clone(pwd()); Pkg.build("AxisAlgorithms"); Pkg.test("AxisAlgorithms"; coverage=true)'
```
Comment (timholy) [inline .travis.yml]: What's the reason for this change?

**Review by timholy — COMMENTED** [review]
Thanks, @carlobaldassi! Looks great to me, I just flagged one spot.

File: .travis.yml (hunk 1 above)
Comment (carlobaldassi) [reply .travis.yml]: I was getting some weird results from codecov reporting a significantly decreased coverage, so this was just a random attempt to see if inlining was the cause. It seems like it didn't work, so I'll ditch this change.

Comment (carlobaldassi) [reply .travis.yml]: ok I reverted this.

**Comment by timholy** [comment]
Thanks!

**Comment by zygmuntszpak** [comment]
This change appears to have broken the `Interpolations.jl` package for Julia 0.6.4.
```julia

 Interpolations                0.7.3
 AxisAlgorithms                0.3.0

Starting Julia...
INFO: Loading HttpServer methods...
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.4 (2018-07-09 19:09 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

Pkg.test("Interpolations")_

WARNING: deprecated syntax "_ as an rvalue".
INFO: Computing test dependencies for Interpolations...
INFO: No packages to install, update or remove
INFO: Testing Interpolations
ERROR: LoadError: LoadError: MethodError: no method matching _A_mul_B_md(::SparseMatrixCSC{Float64,Int64}, ::Array{Float64,2}, ::CartesianRange{CartesianIndex{0}}, ::CartesianRange{CartesianIndex{1}})
Closest candidates are:
  _A_mul_B_md(::AbstractArray{T,2} where T, ::Any, ::Compat.CartesianIndices) at /home/zygmunt/.julia/v0.6/AxisAlgorithms/src/matmul.jl:71
  _A_mul_B_md(::AbstractArray{T,2} where T, ::Any, ::Compat.CartesianIndices, ::Compat.CartesianIndices) at /home/zygmunt/.julia/v0.6/AxisAlgorithms/src/matmul.jl:123
Stacktrace:
 [1] _A_ldiv_B_md!(::Array{Float64,2}, ::WoodburyMatrices.Woodbury{Float64,Base.LinAlg.LU{Float64,Tridiagonal{Float64}},SparseMatrixCSC{Float64,Int64},SparseMatrixCSC{Float64,Int64},Array{Float64,2},Array{Float64,2}}, ::Array{Float64,2}, ::CartesianRange{CartesianIndex{0}}, ::CartesianRange{CartesianIndex{1}}, ::Array{Float64,1}) at /home/zygmunt/.julia/v0.6/Interpolations/src/b-splines/../filter1d.jl:77
 [2] A_ldiv_B_md!(::Array{Float64,2}, ::WoodburyMatrices.Woodbury{Float64,Base.LinAlg.LU{Float64,Tridiagonal{Float64}},SparseMatrixCSC{Float64,Int64},SparseMatrixCSC{Float64,Int64},Array{Float64,2},Array{Float64,2}}, ::Array{Float64,2}, ::Int64, ::Array{Float64,1}) at /home/zygmunt/.julia/v0.6/Interpolations/src/b-splines/../filter1d.jl:15
 [3] prefilter!(::Type{Float64}, ::Array{Float64,2}, ::Type{Tuple{Interpolations.BSpline{Interpolations.Quadratic{Interpolations.Free}},Interpolations.BSpline{Interpolations.Linear}}}, ::Type{Interpolations.OnGrid}) at /home/zygmunt/.julia/v0.6/Interpolations/src/b-splines/prefiltering.jl:74
 [4] prefilter(::Type{Float64}, ::Type{Float64}, ::Array{Float64,2}, ::Type{Tuple{Interpolations.BSpline{Interpolations.Quadratic{Interpolations.Free}},Interpolations.BSpline{Interpolations.Linear}}}, ::Type{Interpolations.OnGrid}) at /home/zygmunt/.julia/v0.6/Interpolations/src/b-splines/prefiltering.jl:47
 [5] interpolate(::Type{Float64}, ::Type{Float64}, ::Array{Float64,2}, ::Tuple{Interpolations.BSpline{Interpolations.Quadratic{Interpolations.Free}},Interpolations.BSpline{Interpolations.Linear}}, ::Interpolations.OnGrid) at /home/zygmunt/.julia/v0.6/Interpolations/src/b-splines/b-splines.jl:73
 [6] anonymous at ./<missing>:?
 [7] include_from_node1(::String) at ./loading.jl:576
 [8] include(::String) at ./sysimg.jl:14
 [9] include_from_node1(::String) at ./loading.jl:576
 [10] include(::String) at ./sysimg.jl:14
 [11] process_options(::Base.JLOptions) at ./client.jl:305
 [12] _start() at ./client.jl:371
while loading /home/zygmunt/.julia/v0.6/Interpolations/test/extrapolation/runtests.jl, in expression starting on line 49
while loading /home/zygmunt/.julia/v0.6/Interpolations/test/runtests.jl, in expression starting on line 8
=====================================[ ERROR: Interpolations ]======================================

failed process: Process(`/media/zygmunt/big/Julia/bin/julia -Cgeneric -J/media/zygmunt/big/Julia/lib/julia/sys.so --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes /home/zygmunt/.julia/v0.6/Interpolations/test/runtests.jl`, ProcessExited(1)) [1]

====================================================================================================
```

**Comment by carlobaldassi** [comment]
@zygmuntszpak the issue with Interpolations [should hopefully be solved now](https://github.com/JuliaLang/METADATA.jl/pull/16188).
